api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+
+/*The context class provides a context identifier*/
+class Context{
```

Comment:
Context also needs to have methods to CreateKey, SetValue, and GetValue 

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/context.md#create-a-key (and the following two sections) which probably means that you'll need an opaque Key object like the Token.

I'm assuming that opentelemetry-cpp/api/include/opentelemetry/common/attribute_value.h is for this use case, so the values should be common::AttributeValue (and so should the ctx_ map).